src/game/AnimalManager.ts

```
@@ -466,155 +478,253 @@ export class AnimalManager {
-      animal.stateTimer = animal.stateDuration;
-      return;
+  private transitionFSM(animal: AnimalInstance, newState: AnimalFsmState): void {
+    const ai = animal.aiData;
+    ai.fsmState = newState;
+    ai.stateTimer = 0;
+
+    switch (newState) {
+      case AnimalFsmState.IDLE:
+        ai.stateDuration = 2 + this.rng.next() * 4;
+        ai.velocity = { x: 0, z: 0 };
+        if (animal.config?.isSkeletal) this.crossfadeTo(animal, animal.config.animations.idle);
```

Biome (2.4.4)

[error] Block statements are preferred in this position.

(lint/style/useBlockStatements)